clients/adapter/src/main/java/org/eclipse/hono/adapter/client/command/InternalCommandSender.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.adapter.client.command;
+
+import org.eclipse.hono.util.Lifecycle;
+
+import io.vertx.core.Future;
+
+/**
+ * A client to send commands to the internal Command and Control API endpoint provided by protocol adapters.
+ */
+public interface InternalCommandSender extends Lifecycle {
+
+    /**
+     * Sends a command to the downstream peer, targeting the protocol adapter Command and Control API endpoint
+     * identified by the given protocol adapter instance id.
+     * <p>
+     * The result of the send operation will lead to one of the {@link CommandContext#accept()},
```

Comment:
FMPOV this needs to be re-phrased to make this a requirement towards implementers, e.g. something like 
> Implementations are required to invoke one of the CommandContext's accept, release, modify or reject methods based on the outcome of sending the command to the protocol adapter instance.

Reply:
As with every interface method contract, it can be seen from both sides. 
I've added the implementers point of view then as well, in the 2nd sentence:

> Note that the result of the send operation is getting applied to the command context here. That means implementations are required to invoke one of the given command context's CommandContext.accept(), CommandContext.release(), CommandContext.modify(boolean, boolean) or CommandContext.reject(String) methods based on the outcome of sending the command to the protocol adapter instance.